services/brig/src/Brig/Data/UserKey.hs

```diff
@@ -164,6 +165,13 @@ deleteKey k = do
   retry x5 $ write deleteHashed (params LocalQuorum (Identity hk))
   retry x5 $ write keyDelete (params LocalQuorum (Identity $ keyText k))
 
+deleteKeyForUser :: (MonadClient m, MonadReader Env m) => UserId -> UserKey -> m ()
+deleteKeyForUser uid k = do
```

Comment:
what problem does this solve?

does it create a new problem in that now, if the key doesn't map onto the uid in half-deleted users any more, we can't clean up the rest?

Reply:
The questions that led to this is: What if a user is partially deleted, their email (or phone number) was deleted correctly and is now used by another user (new account with other `UserId`)?

I think it wouldn't stop another deletion attempt of a partially deleted user, because it doesn't fail (just returns `()`).

Reply:
that makes perfect sense, thanks!  could you add that verbatim as a haddock for future me?